vars/cloudFoundryDeploy.groovy

```diff
@@ -451,38 +451,43 @@ private deploy(def cfApiStatement, def cfDeployStatement, def config, Closure po
             echo "[INFO][${STEP_NAME}] Executing command: '${deployScript}'."
         }
 
-        def returnCode = sh returnStatus: true, script: deployScript
-
-        if(config.verbose || returnCode != 0) {
-            if(fileExists(file: cfTraceFile)) {
-                echo  '### START OF CF CLI TRACE OUTPUT ###'
-                // Would be nice to inline the two next lines, but that is not understood by the test framework
-                def cfTrace =  readFile(file: cfTraceFile)
-                echo cfTrace
-                echo '### END OF CF CLI TRACE OUTPUT ###'
-            } else {
-                echo "No trace file found at '${cfTraceFile}'"
-            }
-        }
+        try {
+            sh deployScript
+        } catch (e) {
+            handleCfCliLog(cfTraceFile)
 
-        if(returnCode != 0){
             error "[${STEP_NAME}] ERROR: The execution of the deploy command failed, see the log for details."
         }
+        if(config.verbose) {
```

Comment:
will print the log twice in case of an error and `config.verbose` true, wouldn't it?
Could maybe improved ...

Reply:
In case of an error we already exit in line 459 on `error`?

Reply:
ah, overlooked this since it was not highlighted in green ...